pkg/cmd/admin/top/imagestreams.go

```diff
@@ -0,0 +1,169 @@
+type TopImageStreamsOptions struct {
+	// internal values
+	Images  *imageapi.ImageList
+	Streams *imageapi.ImageStreamList
+
+	// helpers
+	out      io.Writer
+	osClient client.Interface
+}
+
+// Complete turns a partially defined TopImageStreamsOptions into a solvent structure
+// which can be validated and used for showing limits usage.
+func (o *TopImageStreamsOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, args []string, out io.Writer) error {
+	osClient, _, err := f.Clients()
+	if err != nil {
+		return err
+	}
+	namespace := cmd.Flag("namespace").Value.String()
+	if len(namespace) == 0 {
+		namespace = kapi.NamespaceAll
+	}
+	o.out = out
+
+	allImages, err := osClient.Images().List(kapi.ListOptions{})
+	if err != nil {
+		return err
+	}
+	o.Images = allImages
+
+	allStreams, err := osClient.ImageStreams(namespace).List(kapi.ListOptions{})
+	if err != nil {
+		return err
+	}
+	o.Streams = allStreams
+
+	return nil
+}
+
+// Validate ensures that a TopImageStreamsOptions is valid and can be used to execute command.
+func (o TopImageStreamsOptions) Validate(cmd *cobra.Command) error {
+	return nil
+}
+
+// Run contains all the necessary functionality to show current image references.
+func (o TopImageStreamsOptions) Run() error {
+	infos := o.imageStreamsTop()
+	Print(o.out, ImageStreamColumns, infos)
+	return nil
+}
+
+var ImageStreamColumns = []string{"NAME", "STORAGE", "IMAGES", "LAYERS"}
+
+// imageStreamInfo contains contains statistic information about ImageStream usage.
+type imageStreamInfo struct {
+	ImageStream string
+	Storage     int64
+	Images      int
+	Layers      int
+}
+
+var _ Info = &imageStreamInfo{}
+
+func (i imageStreamInfo) PrintLine(out io.Writer) {
+	printValue(out, i.ImageStream)
+	printSize(out, i.Storage)
+	printValue(out, i.Images)
+	printValue(out, i.Layers)
+}
+
+// imageStreamsTop generates ImageStream information from a graph and
+// returns this as a list of imageStreamInfo array.
+func (o TopImageStreamsOptions) imageStreamsTop() []Info {
+	g := graph.New()
+	addImagesToGraph(g, o.Images)
+	addImageStreamsToGraph(g, o.Streams)
+
+	infos := []Info{}
+	streamNodes := getImageStreamNodes(g.Nodes())
+	for _, sn := range streamNodes {
+		storage, images, layers := getImageStreamSize(g, sn)
+		infos = append(infos, imageStreamInfo{
+			ImageStream: fmt.Sprintf("%s/%s", sn.ImageStream.Namespace, sn.ImageStream.Name),
+			Storage:     storage,
+			Images:      images,
+			Layers:      layers,
+		})
+	}
+
+	return infos
+}
+
+func getImageStreamSize(g graph.Graph, node *imagegraph.ImageStreamNode) (int64, int, int) {
+	imageEdges := g.OutboundEdges(node, ImageStreamImageEdgeKind)
+	storage := int64(0)
+	images := len(imageEdges)
+	layers := 0
+	for _, e := range imageEdges {
+		imageNode, ok := e.To().(*imagegraph.ImageNode)
+		if !ok {
+			continue
+		}
+		image := imageNode.Image
+		layers += len(image.DockerImageLayers)
+		for _, layer := range image.DockerImageLayers {
+			storage += layer.LayerSize
```

Comment:
Please see [here](https://github.com/openshift/origin/blob/master/pkg/image/api/helper.go#L511) for counting size of unique layers. Don't count one layer more than once.